os/arch/arm/src/armv7-a/arm_addrenv.c

```diff
@@ -543,124 +517,95 @@ ssize_t up_addrenv_heapsize(const group_addrenv_t *addrenv)
+		}
+
+		/* Set (or clear) the new page table entry */
+
+		paddr = (uintptr_t) addrenv->text[i];
+		if (paddr) {
+			binfo("text: set l1 entry (paddr=%x vaddr=%x)\n", paddr, vaddr);
+			mmu_l1_setentry(paddr, vaddr, MMU_L1_PGTABFLAGS);
+		} else {
+			binfo("text: clear l1 (vaddr=%x)\n", vaddr);
+			mmu_l1_clrentry(vaddr);
+		}
+	}
+
+	for (vaddr = CONFIG_ARCH_DATA_VBASE, i = 0; i < ARCH_DATA_NSECTS; vaddr += SECTION_SIZE, i++) {
+		/* Save the old L1 page table entry */
+
+		if (oldenv) {
+			oldenv->data[i] = mmu_l1_getentry(vaddr);
+		}
+
+		/* Set (or clear) the new page table entry */
+
+		paddr = (uintptr_t) addrenv->data[i];
+		if (paddr) {
+			binfo("data: set l1 entry (paddr=%x vaddr=%x)\n", paddr, vaddr);
+			mmu_l1_setentry(paddr, vaddr, MMU_L1_PGTABFLAGS);
+		} else {
+			binfo("data: clear l1 (vaddr=%x)\n", vaddr);
+			mmu_l1_clrentry(vaddr);
+		}
+	}
 
 #ifdef CONFIG_BUILD_KERNEL
-  for (vaddr = CONFIG_ARCH_HEAP_VBASE, i = 0;
-       i < ARCH_HEAP_NSECTS;
-       vaddr += SECTION_SIZE, i++)
-    {
-      /* Save the old L1 page table entry */
-
-      if (oldenv)
-        {
-          oldenv->heap[i] = mmu_l1_getentry(vaddr);
-        }
-
-      /* Set (or clear) the new page table entry */
-
-      paddr = (uintptr_t)addrenv->heap[i];
-      if (paddr)
-        {
-          binfo("heap: set l1 entry (paddr=%x vaddr=%x)\n", paddr, vaddr);
-          mmu_l1_setentry(paddr, vaddr, MMU_L1_PGTABFLAGS);
-        }
-      else
-        {
-          binfo("data: clear l1 (vaddr=%x)\n", vaddr);
-          mmu_l1_clrentry(vaddr);
-        }
-    }
+	for (vaddr = CONFIG_ARCH_HEAP_VBASE, i = 0; i < ARCH_HEAP_NSECTS; vaddr += SECTION_SIZE, i++) {
+		/* Save the old L1 page table entry */
+
+		if (oldenv) {
+			oldenv->heap[i] = mmu_l1_getentry(vaddr);
+		}
+
+		/* Set (or clear) the new page table entry */
+
+		paddr = (uintptr_t) addrenv->heap[i];
+		if (paddr) {
+			binfo("heap: set l1 entry (paddr=%x vaddr=%x)\n", paddr, vaddr);
+			mmu_l1_setentry(paddr, vaddr, MMU_L1_PGTABFLAGS);
+		} else {
+			binfo("data: clear l1 (vaddr=%x)\n", vaddr);
+			mmu_l1_clrentry(vaddr);
+		}
+	}
 
 #ifdef CONFIG_MM_SHM
-  for (vaddr = CONFIG_ARCH_SHM_VBASE, i = 0;
-       i < ARCH_SHM_NSECTS;
-       vaddr += SECTION_SIZE, i++)
-    {
-      /* Save the old L1 page table entry */
-
-      if (oldenv)
-        {
-          oldenv->shm[i] = mmu_l1_getentry(vaddr);
-        }
-
-      /* Set (or clear) the new page table entry */
-
-      paddr = (uintptr_t)addrenv->shm[i];
-      if (paddr)
-        {
-          mmu_l1_setentry(paddr, vaddr, MMU_L1_PGTABFLAGS);
-        }
-      else
-        {
-          mmu_l1_clrentry(vaddr);
-        }
-    }
+	for (vaddr = CONFIG_ARCH_SHM_VBASE, i = 0; i < ARCH_SHM_NSECTS; vaddr += SECTION_SIZE, i++) {
+		/* Save the old L1 page table entry */
+
+		if (oldenv) {
+			oldenv->shm[i] = mmu_l1_getentry(vaddr);
+		}
+
+		/* Set (or clear) the new page table entry */
+
+		paddr = (uintptr_t) addrenv->shm[i];
```

Comment:
Could you search `_t) `? Then you can find 29 lines.

Reply:
ok